src/components/new-safe/create/steps/ReviewStep/index.tsx

```diff
@@ -151,6 +166,41 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
       saltNonce: saltNonce.toString(),
     }
 
+    if (isCounterfactual) {
+      const counterfactualNonce = await getCounterfactualNonce(provider, { ...props, saltNonce: '0' })
```

Comment:
If we're switching to an incremental nonce, I would do it for all deployments, not just counterfactual. For the benefitsts that Micah outlined and for consistency.

Reply:
It would require us to check on-chain if a contract is already deployed to that address for every safe that is created but I think it would simplify our code so I will look into refactoring this part of safe creation a bit more.